anyway/infographics_utils.py

```diff
@@ -1042,73 +1010,54 @@ def pedestrian_injured_in_junctions_mock_data():  # Temporary for Frontend
         ]
 
 
+create_widget_dict = {
+    WidgetId.accident_count_by_severity: AccidentCountBySeverityWidget,
+    WidgetId.most_severe_accidents_table: MostSevereAccidentsTableWidget,
+    WidgetId.most_severe_accidents: MostSevereAccidentsWidget,
+    WidgetId.street_view: StreetViewWidget,
+    WidgetId.head_on_collisions_comparison: HeadOnCollisionsComparisonWidget,
+    WidgetId.accident_count_by_accident_type: AccidentCountByAccidentTypeWidget,
+    WidgetId.accidents_heat_map: AccidentsHeatMapWidget,
+    WidgetId.accident_count_by_accident_year: AccidentCountByAccidentYearWidget,
+    WidgetId.injured_count_by_accident_year: InjuredCountByAccidentYearWidget,
+    WidgetId.accident_count_by_day_night: AccidentCountByDayNightWidget,
+    WidgetId.accident_count_by_hour: AccidentCountByHourWidget,
+    WidgetId.accident_count_by_road_light: AccidentCountByRoadLightWidget,
+    WidgetId.top_road_segments_accidents_per_km: TopRoadSegmentsAccidentsPerKmWidget,
+    WidgetId.injured_count_per_age_group: InjuredCountPerAgeGroupWidget,
+    WidgetId.vision_zero: VisionZeroWidget,
+    WidgetId.accident_count_by_driver_type: AccidentCountByDriverTypeWidget,
+    WidgetId.accident_count_by_car_type: AccidentCountByCarTypeWidget,
+    WidgetId.injured_accidents_with_pedestrians: InjuredAccidentsWithPedestriansWidget,
+    WidgetId.accident_severity_by_cross_location: AccidentSeverityByCrossLocationWidget,
+    WidgetId.motorcycle_accidents_vs_all_accidents: MotorcycleAccidentsVsAllAccidentsWidget,
+    WidgetId.accident_count_pedestrians_per_vehicle_street_vs_all: AccidentCountPedestriansPerVehicleStreetVsAllWidget,
+    WidgetId.top_road_segments_accidents: TopRoadSegmentsAccidentsWidget,
+    WidgetId.pedestrian_injured_in_junctions: PedestrianInjuredInJunctionsWidget,
+}
```

Comment:
Sorry to pick on this again, but I think there are other approaches to doing this, that we should at least consider.

My favorite, replacing both this part and the argument in the `super()` calls, is using a `register` decorator, to be used like this:
```python
@widget_collection.register
class TopRoadSegmentsAccidentsWidget(Widget):
    def __init__(self): # nothing dedicated here
```

Then, instead of `create_widget()`, we'll simply iterate over `widget_collection`.

This can also be done using a metaclass for `Widget`, without anything explicit in the inheriting classes, but I'd prefer the explicit way, using a decorator. If we want to keep the enum, we can pass the appropriate value to `widget_collection.register`. But I don't see why we need it.

Here's a proposed implementation (untested):
```python
class WidgetCollection:
    widgets: List[Type[Widget]]
    def __init__(self):
        self.widgets = []
    def register(self, widget_class: Type[Widget]) -> Type[Widget]:
        widgets.append(widget_class)
        return widget_class

widget_collection = WidgetCollection()
```

Reply:
It can even be made simpler, using a global variable `registered_widgets` and a global function similar to `register` defined above.